### PR TITLE
split 'render' support into '2d' and '3d'

### LINKED ISSUE #31: Split `render` cargo feature into 2d and 3d
I like to selectively depend on either `bevy_sprite` or `bevy_pbr` depending on whether the project in question is a 2d or 3d game, to not include unnecessary parts of Bevy into my projects. Currently, `bevy_asset_loader` requires pulling in both. Would be nice if the two were separate.

### PR DESCRIPTION
**Pull request by inodentry**
Greatly improves the usability of bevy features configuation.

Support for texture atlases and standard materials split into separate `2d` and `3d` features.

Now it also ensures that the appropriate bevy features are enabled.

Closes #31

### CODE REVIEW

File: bevy_asset_loader/Cargo.toml [hunk 1]
```diff
@@ -13,8 +13,16 @@ categories = ["game-development"]
 readme = "../README.md"
 
 [features]
-# This feature requires bevy's TextureAtlas and StandardMaterial assets to be available ("bevy/render")
-render = ["bevy_asset_loader_derive/render"]
+# This feature adds support for bevy's TextureAtlas assets
+2d = [
+    "bevy/bevy_sprite",
```
Comment (NiklasEi) [inline bevy_asset_loader/Cargo.toml:18]: Do we actually need the Bevy features?

Comment (inodentry) [reply bevy_asset_loader/Cargo.toml:18]: Well, I don't know (haven't tested) if we *strictly need* them (as in, can we compile without them), but the relevant functionality that our features enable is kinda useless without them anyway. I think there shouldn't be a situation where the user wants to enable the `2d`/`3d` on this crate, but disable the bevy features. I decided to add them here to improve the UX, to ensure bevy is always built appropriately / the user hasn't forgotten about them.

Comment (NiklasEi) [reply bevy_asset_loader/Cargo.toml:18]: Sounds good to me :+1:

**Comment by NiklasEi** [comment]
The builds are currently red, because `bevy_asset_loader/tests/ui/missing_feature.stderr` contains the old feature names.

**Review by NiklasEi — COMMENTED** [review]
The correct features are missing on https://github.com/NiklasEi/bevy_asset_loader/blob/main/bevy_asset_loader/src/dynamic_asset.rs#L27 and https://github.com/NiklasEi/bevy_asset_loader/blob/main/bevy_asset_loader/src/dynamic_asset.rs#L33

**Comment by inodentry** [comment]
When I grep for `feature.*render` i find a bunch of other places, specifically in your tests. I don't understand how your tests work, you seem to have some tests marked for having vs not having the render feature and I don't understand why.

Could you please take care of fixing your tests by yourself? I'm having a hard time with them.

**Comment by NiklasEi** [comment]
Fixed in IyesGames/bevy_asset_loader#1

**Review by NiklasEi — APPROVED** [review]

**Pull request merged**